multisig-wallet/project/contracts/multisig-contract/src/main.sw

```diff
@@ -56,67 +83,125 @@ impl MultiSignatureWallet for Contract {
 
     #[storage(read, write)]
     fn execute_transaction(
-        data: b256,
+        contract_call_params: Option<ContractCallParams>,
```

Comment:
Yh, these call / transfer params can probably enum'd to and typed to remove optionals. This in addition to `new()` would clean this up

Reply:
The enum suggestion would not work as a transfer does not require `contract_call_params`, but a call _may_ include `transfer params`.